package/boot/grub2/Makefile

```diff
@@ -71,51 +76,73 @@ HOST_CONFIGURE_ARGS += \
 	--sbindir="$(STAGING_DIR_HOST)/bin" \
 	--disable-werror \
 	--disable-libzfs \
-	--disable-nls
+	--disable-nls \
+	--with-platform=none
 
 HOST_MAKE_FLAGS += \
 	TARGET_RANLIB=$(TARGET_RANLIB) \
 	LIBLZMA=$(STAGING_DIR_HOST)/lib/liblzma.a
 
+TARGET_CFLAGS :=
+
 define Host/Configure
 	$(SED) 's,(RANLIB),(TARGET_RANLIB),' $(HOST_BUILD_DIR)/grub-core/Makefile.in
 	$(Host/Configure/Default)
 endef
 
-define Host/Install
-	$(call Host/Install/Default)
-
-	$(INSTALL_DIR) $(STAGING_DIR_HOST)/lib/grub/grub2-generic
+define Package/grub2/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/grub-bios-setup $(1)/usr/sbin/
+	$(INSTALL_DIR) $(STAGING_DIR_IMAGE)/grub2
+	$(CP) $(PKG_BUILD_DIR)/grub-core/boot.img $(STAGING_DIR_IMAGE)/grub2/
+	$(CP) $(PKG_BUILD_DIR)/grub-core/cdboot.img $(STAGING_DIR_IMAGE)/grub2/
+	sed 's#msdos1#gpt1#g' ./files/grub-early.cfg >$(PKG_BUILD_DIR)/grub-early.cfg
+	$(STAGING_DIR_HOST)/bin/grub-mkimage \
+		-d $(PKG_BUILD_DIR)/grub-core \
+		-p /boot/grub \
+		-O i386-pc \
+		-c $(PKG_BUILD_DIR)/grub-early.cfg \
+		-o $(STAGING_DIR_IMAGE)/grub2/gpt-core.img \
+		at_keyboard biosdisk boot chain configfile fat linux ls part_gpt reboot serial vga
 	$(STAGING_DIR_HOST)/bin/grub-mkimage \
-		-d $(STAGING_DIR_HOST)/lib/grub/i386-pc \
+		-d $(PKG_BUILD_DIR)/grub-core \
 		-p /boot/grub \
 		-O i386-pc \
 		-c ./files/grub-early.cfg \
-		-o $(STAGING_DIR_HOST)/lib/grub/grub2-generic/core.img \
+		-o $(STAGING_DIR_IMAGE)/grub2/generic-core.img \
 		at_keyboard biosdisk boot chain configfile ext2 linux ls part_msdos reboot serial vga
-
-	$(INSTALL_DIR) $(STAGING_DIR_HOST)/lib/grub/grub2-iso
 	$(STAGING_DIR_HOST)/bin/grub-mkimage \
-		-d $(STAGING_DIR_HOST)/lib/grub/i386-pc \
+		-d $(PKG_BUILD_DIR)/grub-core \
 		-p /boot/grub \
 		-O i386-pc \
 		-c ./files/grub-early.cfg \
-		-o $(STAGING_DIR_HOST)/lib/grub/grub2-iso/eltorito.img \
-		at_keyboard biosdisk boot chain configfile iso9660 linux ls part_msdos reboot serial vga
-
-	$(INSTALL_DIR) $(STAGING_DIR_HOST)/lib/grub/grub2-legacy
+		-o $(STAGING_DIR_IMAGE)/grub2/eltorito.img \
+		at_keyboard biosdisk boot chain configfile iso9660 linux ls part_msdos reboot serial test vga
 	$(STAGING_DIR_HOST)/bin/grub-mkimage \
-		-d $(STAGING_DIR_HOST)/lib/grub/i386-pc \
+		-d $(PKG_BUILD_DIR)/grub-core \
 		-p /boot/grub \
 		-O i386-pc \
 		-c ./files/grub-early.cfg \
-		-o $(STAGING_DIR_HOST)/lib/grub/grub2-legacy/core.img \
+		-o $(STAGING_DIR_IMAGE)/grub2/legacy-core.img \
 		biosdisk boot chain configfile ext2 linux ls part_msdos reboot serial vga
 endef
 
-define Package/grub2/install
-	$(INSTALL_DIR) $(1)/usr/sbin
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/grub-bios-setup $(1)/usr/sbin/
+define Package/grub2-efi/install
+	sed 's#msdos1#gpt1#g' ./files/grub-early.cfg >$(PKG_BUILD_DIR)/grub-early.cfg
```

Comment:
This package ends up empty, perhaps you've forget to add some files in it?

Reply:
There are no files should be in this package.